[PVR] API 3.0.0 Changes

metaron-uk opened the pull request:
Several _"good ideas"_ which require a PVR API version bump have been discussed on the forum (http://forum.kodi.tv/showthread.php?tid=227026).

This PR groups these API changes together (one commit for each _"good idea"_) with the actual PVR code changes which implement them to be submitted as separate (referenced) PRs.

Once the API changes are considered 'acceptable' and 'necessary', this change can be merged so PVR addon developers don't have to roll their API includes several times in quick succession.

Supporting code changes can be discussed in the separate PRs without swamping the API discussion.
Supporting changes could even be merged after this PR if they are not quite ready at the time.

metaron-uk committed:
[PVR] API 3.0.0 Version Bump

metaron-uk committed:
[PVR] API: Add boolean control of Any Time

Re-name PVR_TIMER_TYPE_SUPPORTS_START_END_TIME -> PVR_TIMER_TYPE_SUPPORTS_START_TIME
Add Start and End AnyTime control booleans and associated 'SUPPORTS' flags to API
Add PVR_TIMER_TYPE_SUPPORTS_END_TIME flag to API

metaron-uk committed:
[PVR] API: Add Recording scraper support

Add iSeriesNumber, iEpisodeNumber, strEpisodeName and iYear to API
Document their intended use in improving recording scraper and UI integration

metaron-uk committed:
[PVR] Implement: Add boolean control of Any Time

Allow Any Time to be toggled on/off without affecting start/end clocks
Replace startTime=endTime=0 with booleans bStartAnyTime and bEndAnyTime
Remove functions and variables required for the old method
Add short strSummary format if end is anytime
Don't use FirstDay in strSummary as startTime can now remain valid

metaron-uk commented:
@ksooo - API change PR submitted with some of your feedback incorporated. 
Any further comments?

ksooo commented:
@opdenkamp we need your assistance and guidance on this, please. There are several points in here I'm more than uncertain about.

Jalle19 commented:
Since we're changing the API again I'd like to include some extensions to the signal status handling so the addon can provide stuff like number of tuners, current tuner number, current delivery system etc.

I'll reply once I have something.

metaron-uk commented:
@Jalle19 - These should fix the whitespace issues you identified. Will squash later on rebase.

metaron-uk committed:
[PVR] API: Add maxRecordings

Adds API support for an (optional) list to limit a timer's number of recordings

metaron-uk commented:
Rebased to incorporate some of the feedback and make this a major version bump.
Re-ordered commits in order of concept maturity (imo).

metaron-uk commented:
Dropped sortTime from the list of potential changes. IMO the new AnyTime management which allows startTime to remain a sensible value is sufficient.

metaron-uk committed:
[PVR] Implement: Recording scraper support / UI

Add recording as source for EpisodeName skinnable value
Add recording as source for Season/Episode/Year skinnable values
Confluence skin changes to display recording EpisodeName if available
pvr:// filename now includes season/episode, year and EpisodeName:
 - UI Sort 'By File' now displays in season/episode order
 - fixes ttvdb scraping of tv shows using pvr:// protocol

metaron-uk commented:
I've decided to drop eda5c37 ([PVR] Add strEpisodeName to API) from the list of API3.0.0 changes for the following reasons:

1. EpisodeName is already accessible from the EPGInfoTag associated with a child timer (aka upcoming recording) as long as it is within Kodi's currently requested EPG range.
2. If you display EpisodeName in the timer list for a child timer, when you access information on that timer you loose the EpisodeName display as there is no place to display it in the Timer Settings dialog.

IMO it is better for the addon rather than the skin to append the Episode Name to the child timer's 'Title' in this case. As an 'upcoming recording' EPG based timer will be read-only anyhow, I don't see this as an issue.

The only slight problem is the format used to add Episode Name will be set by the client, not the skin: 
Confluence currently uses "TITLE (EPISODENAME)", pvr.mythtv currently uses "TITLE - EPISODENAME"

Maybe this format could be localized so both skins and clients could use it, but I consider this out of scope for a PVR API change.

ryangribble committed:
[PVR] Implement: Add menu visibility controls

Prevent timers declaring they require an EPG tag appearing below 'Add...'
Prevent timers declaring they forbid an EPG tag appearing below 'Add Customer Timer'
Compatibility code menu visibility adjustments

ryangribble committed:
[PVR] Implement: MaxRecordings list

Optional list limiting number of recordings associated with a timer
Appears directly below Lifetime in the Timer Settings dialog if enabled
Intended use: +ve values=number of recordings, -ve values=addon defined

metaron-uk committed:
[PVR] API: Add menu visibility controls for timers

Add flags which can be used to prevent timer types:
1) needing an EPG Tag from appearing in 'Add...' menu
2) not related to EPG Tags appearing in 'Add Custom Timer' list

metaron-uk committed:
[PVR] Doc Only: iLifetime list purpose

metaron-uk committed:
[PVR] Doc Only: Doxy syntax correction

metaron-uk commented:
OK, I'm going to drop the last three proposed commits from this PR:

f6b99fd @metaron-uk 	[PVR] Add strRecordingId to EPG for API
strRecordingId was removed from the interface not that long ago and I can see good reasons for doing so. Really what's needed here to fulfil my intended purpose is 'ProgramID' (i.e. a unique ID provided by the listings source, or maybe better ttvdb/imdb), both for the EPG entry and the Recording - these could then be used to match up EPG entries with recorded programs, regardless of source (maybe even shows in the video library?). Whatever the solution ends up being, it needs more time to think it through.

ae57974 @metaron-uk 	[PVR] Add GetRecordingEpgInfo f'n to API 
ttvdb and imdb scraper support is included in this PR, so a consistent source can be used to categorize downloaded and recorded programmes. IMO the 'Recordings' view already provides sufficient metadata scraped by the backend for Jarvis. Again more thought and time would be required to prototype a propper solution.

848c0c5 @metaron-uk 	[PVR] Add GetTimerEpgInfo f'n to API
I'd still like to add a context menu option in the timer view to 'Show EPG Data' on an upcoming recording (where available), but it seems reasonable to restrict this facility to the Kodi EPG data range.
If you want more data (and your hardware can handle it), simply extend the EPG range covered.

metaron-uk commented:
I'm now pulling all my and @ryangribble 's commits together on a new branch and starting the process of creating the client addon support PRs for this change. 

@Jalle19 - you said there was another API change you wanted to get in at the same time. Is this ready?

**Timescale Objective**
@ryangribble has said he wants two of these changes for wmc asap and I'd like to get it merged before 20th August when my availability will decline drastically. @ksooo has said he won't be available much from the 7th August to the 20th so I'd like to have addressed all the issues to his satisfaction by the 7th.

**Way Forward**
My feeling is that I should now close this PR (and the other related ones) and raise a new one with all the changes for final review/comment and merge. I'll probably do this tomorrow. 
*Please shout if there are any concerns with this approach*.

**Outstanding Issues**
To my mind there are now only two outstanding comments raised on this package of changes:

1. Doubt about the usefulness of the 'scraper' support changes expressed by @ksooo and @ryangribble (seems they are both in areas of the world with poor support for season/episode number in EPG data). I've done the legwork and find it extremely useful myself here in the UK, so I'm keeping it in for now. +1's would be appreciated :).

2. A possible issue with the 'requirements' for Lifetime and MaxRecordings. @ksooo expressed unease that Lifetime / MaxRecordings won't necessarily be implemented in a consistent manner across all clients, which might make relying on the value of these settings in Kodi core difficult in future. @ryangribble and I see this as necessary flexibility to support different backends. 
Hopefully this will just be a case of minor adjustments to the doxy for these (and maybe other) optional timer fields. If a specific use case or embryonic idea exists which the current approach breaks, maybe a fix can be found?

If I 'forgot' some, I'm sure you will remind me...

ksooo commented:
Regarding next steps:
IMO it is not necessary to create a new PR. Just use this one.

Regarding the issues:

Re 1: My doubt here come from the knowledge that tvheadend today lacks support for this, not from  the fact that I live in a country with poor support for epg (which is actually not the case) and therfore don't care. If there is at least one backend that supports this stuff I'm fine with the API extension. Having no knowledge about how for instance numbering of seasons and episodes works (is there some standard, how does Kodi handle things, etc) I cannot do a code review (like I already said). This, my +1 if someone with enough knowledge does a code review and gives his +1.

Re 2: Again my -1 for the definition of semantics of "max recordings" and the changes done here for "lifetime" (yes, it's all just about API documentation, guys!). To weak, for no valid reason. I'm not going to explain things again. Already did it in #7701. My proposal gives you imo 100% the functional freedeom you need. You are only loosing (maybe) the most easiest way to implement stuff in your addons as you cannot just route the backend values to Kodi and back but may need extra mapping. OTOH, your proposal to define nothing will rob Kodi the chance to do something meaningful with the data now or in the future.

metaron-uk commented:
Thanks @ksooo. I'll go through your #7701 comments again tomorrow and see if I can address them properly. Got an early start tomorrow so need an early night :-)

ryangribble commented:
+1 for me on the scraper stuff.  I have no problems with it and will populate any of the fields I can from WMC eventhough its not something useful in my particular country

Regarding the 2nd point, discussion is already happening on #7701 but fundamentally i will do whatever is required to get this into Kodi 16, I just want to ensure we have agreement/consensus and part of that is understanding why.  I also need to go over the latest discussion there

Jalle19 commented:
@metaron-uk sorry, not ready yet.

metaron-uk commented:
@Jalle19 - no problem.
You don't happen to live in an area with good EPG (series/episode data) and use a mythtv or wmc pvr backend to you? 
Tvheadend doesn't support this data for recordings so @ksooo wanted someone else to code review / check my 'scrapers' fix ( #7677 )

ksooo commented:
I think @Jalle19 was talking about his additional API change request.;-)

metaron-uk commented:
Just being cheeky - he can always say no....

metaron-uk commented:
Rebased on current kodi master with changes cherry-picked from the related PRs (to be closed).
API changes are prefixed with "API:". Implementation changes with "Implement:" etc...

**PVR Client Changes**
** *Minimal "maintains existing functionality" PRs will be merged to the kodi addon repositories. Changes are only necessary if you want to take full advantage of the new features* **

*Anytime*
Where an addon previously set ```startTime=endTime=0``` to indicate Any Time, this should be replaced by setting ```bStartAnyTime``` and/or ```bEndAnyTime```.
Where ```startTime == 0 && endTime == 0``` was used to control backend settings, this should now be based on ```bStartAnyTime``` and/or ```bEndAnyTime```
```PVR_TIMER_SUPPORTS_ANYTIME``` has been replaced with  ```PVR_TIMER_SUPPORTS_START_ANYTIME``` and ```PVR_TIMER_SUPPORTS_END_ANYTIME```.
```PVR_CLIENT_SUPPORTS_START_END``` has been replaced with ```PVR_CLIENT_SUPPORTS_START``` and ```PVR_CLIENT_SUPPORTS_END```.

*Scraper / Recording Episode Name*
No changes are required to clients for them to function. Any clients using the "TV Show" strPlotSummary approach will see a debug message that this has been deprecated. Information extracted from strPlotSummary will be used to populate EpisodeName.
Any clients which are currently merging EpisodeName into Title will need to provide it and season/episode/year separately for the scraper and list sorting improvements to work.
If your backend does not provide season/episode information for recordings, but only EpisodeName, you won't see scraper or sorting improvements.

*Menu Visibility Controls*
By default all timer types (other than ```PVR_TIMER_TYPE_FORBIDS_NEW_INSTANCES```) will now appear in both 'Add...' and 'Add Customer Timer' menus unless using the compatibility code.
To remove a timer type from one of these menus set ```PVR_TIMER_TYPE_FORBIDS_EPG_TAG_ON_CREATE``` and/or ```PVR_TIMER_TYPE_REQUIRES_EPG_TAG_ON_CREATE```.

*Max Recordings*
New (optional) feature allowing a client to present a 'Max Recordings' selection list. +ve values: number of recordings, -ve values: addon defined. Appears directly below 'Lifetime' if enabled. No change to addon = disabled.

**Known Issues**
*Lifetime*
pvr.wmc and pvr.mythtv current implementations use iLifetime as 'Recording Lifetime'. pvr.hts uses iLifetime as 'Timer Lifetime'. Doxy changed to make the purpose of iLifetime ambiguous to cover both, however given @ksooo's previous comments this will probably be unsatisfactory.
Options would appear to be
1. live with this inconsistency (no further change required)
2. use mututally exclusive ```PVR_TIMER_TYPE_SUPPORTS_TIMER_LIFETIME``` and ```PVR_TIMER_TYPE_SUPPORTS_RECORDING_LIFETIME``` flags which use the same existing 'Lifetime' variables
3. Adjust the code to create a new 'Recording Lifetime' list. Adjust those addons which use the 'other' method.

*Scraper*
imdb doesn't seem to honour the 'year' information when scraping recordings.
@ksooo has requested someone else to review this change as tvheadend doesn't support season/episode on recordings

*Addon PRs*
@metaron-uk will do the change for pvr.mythtv
@ryangribble will do the change for pvr.wmc
The other PRs will be shared out between them. 

Any offers of assistance / comments / suggestions / constructive criticism gratefully received - preferably sooner rather than later.

Jalle19 commented:
Just to make it clear, I'm not blocking anything just because I haven't had time to code myself :-P

metaron-uk commented:
@ryangribble - I'm calling it a night, feel free to continue where I left off (I'll be back in ~10 hours or so)
The quick win 'compatibilty mode' addons are relatively easy to spot - search for PVR_TIMER in src/client.cpp

opdenkamp commented:
will review this some evening this week. house building is taking a bit 
longer than planned (as usual) and is taking most of my time off atm.

On 03-08-15 22:39, metaron-uk wrote:
> @ryangribble <https://github.com/ryangribble> - I'm calling it a night,
> feel free to continue where I left off (I'll be back in ~10 hours or so)
> The quick win 'compatibilty mode' addons are relatively easy to spot -
> search for PVR_TIMER in src/client.cpp
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xbmc/xbmc/pull/7626#issuecomment-127401017>.
>

metaron-uk commented:
@ksooo - I've done all the 'quick win' addon PRs and one for mythtv. @ryangribble has promised to do a 'minimal' pvr.wmc change. 
Unless you have objections, I'll now have a go at a 'minimal support' change for pvr.hts (I have a tvheadend server installation and a 'spare' USB tuner, so should be able to test it properly)

@hudokkow has already merged a change to make kodi core use 'master' branch rather than a specific commit hash for addons, so I'm guessing that's it? Let me know if something else is still missing.

**Lifetime**
As it looks like only 3 addons have actually implemented the new API so far, I'd be happy to add an extra change to this PR to address the 'lifetime' issue. I can then update the 'minimal support' PR for whichever addon(s) use the 'incorrect' method if necessary.
Another option is introducing the three 'addon defined options' you proposed and using one of them for the 'other' implementation. If you want me to go this route I'd slightly prefer to adapt pvr.hts (with your buy-in) than wmc and mythtv (which would need me to get buy-in from @janbar and @ryangribble).

Alternatively this PR doesn't actually have to deal with the 'lifetime' issue at all. I could simply remove the lifetime doxy commit and defer the issue. However, IMO this is the last 'easy' chance to make changes to this part of the API before more than 3 clients are using it. If possible I'd like to resolve this to everyone's satisfaction now.

Just let me know which option you think is best and I'll get on to it.

metaron-uk commented:
@opdenkamp - brilliant!

ksooo commented:
@metaron-uk go ahead, change pvr.hts.

metaron-uk commented:
@ksooo: Addressed your comments and squashed the fixups. Let me know if you think of anything else.

ksooo commented on xbmc/addons/include/xbmc_pvr_types.h:
```diff
@@ -388,7 +404,8 @@ extern "C" {
     int             iPriority;                                 /*!< @brief (optional) the priority of this timer */
-    int             iLifetime;                                 /*!< @brief (optional) lifetime of this timer in days. After this time period, the recording shall be automatically deleted by the backend */
+    int             iLifetime;                                 /*!< @brief (optional) lifetime of recordings created by this timer. > 0 days after which recordings will be deleted by the backend, < 0 addon defined integer list reference, == 0 disabled */
+    int             iMaxRecordings;                            /*!< @brief (optional) integer ref to list of maximum recordings: > 0 number of recordings, < 0 addon defined, == 0 disabled */
```
Almost. Could we please make the two comments  a little more identical, wrt. wording, punctuation etc? Lifetime is IMO fine already, for max recordings:

"maximum number of recordings this timer shall create. > 0 number of recordings, < 0 addon defined integer list reference, == 0 disabled"

metaron-uk replied:
done

ksooo commented:
@metaron-uk thanks for your patience. +1 from me for the whole bunch of commits, except the implementation of the "recording scraper stuff", because of lack of knowledge on my side. ;-)

ksooo commented:
.. ah and ofc I have not yet delt with the addon PRs. That's another story.

metaron-uk commented:
Right, just waiting for the pvr.wmc PR and the inevitable *feedback*. I'm taking the evening off.

ryangribble commented:
Ive created the minimal support PR for pvr.wmc.  No changes were required except version bump etc.

I will submit my full series recording implementation for pvr.wmc soon this week, but this minimal PR means that all the API merge can go ahead etc regardless of my final implementation

metaron-uk commented:
I did a quick grep on the other addons. None of them appear to have a ClientVersion or GetClientVersion method like pvr.wmc does. If anyone knows different, please let me know so I can fix the appropriate PR.

ryangribble commented:
I haven't seen any other addons with a compiled in version number. It's been such a hassle though as almost nobody remembers to bump it 
I wish I could generate it as part of the build 
Or maybe I'll just give up and read it from the addon.xml file (which just seems wrong since you could claim to be a different version just by editing the XML file)

Glenn-1990 commented:
Known Issues
Lifetime
pvr.wmc and pvr.mythtv current implementations use iLifetime as 'Recording Lifetime'. pvr.hts uses iLifetime as 'Timer Lifetime'. Doxy changed to make the purpose of iLifetime ambiguous to cover both, however given @ksooo's previous comments this will probably be unsatisfactory.
Options would appear to be
1. live with this inconsistency (no further change required)

@metaron-uk Is this an issue? The lifetime setting in pvr.hts is defining how long it's recording must be kept (not the timer). So it's similar than wmc and mythtv..

ksooo commented:
> pvr.hts uses iLifetime as 'Timer Lifetime'.

This is not the case. pvr.hts uses lifetime as recording lifetime.

metaron-uk commented:
I guessed it was something like that. 
The doxy was just 'wrong' and now is 'correct' for all three implemented clients. (phew!)

ksooo commented:
@metaron-uk Exactly.

janbar commented:
...is exactly recording lifetime
@ksooo then why not just rename the attribute as it is instead add a new one ?

metaron-uk commented:
@janbar, that's exactly what we did (a131127) - but only the comments needed to change.

janbar commented:
Sorry, i have seen ..TIMER_LIFETIME in one commit. Probably discarded since :8ball:

metaron-uk commented:
**Known Issues**
*Scraper*
imdb doesn't seem to honour the 'year' information when scraping recordings.

I've investigated further and it appears that the scraper *is* being provided with 'year', but my listings provider uses 'imdb year' rather than themoviedb year e.g. '300' is listed as 300 (2006) not 300 (2007).
Nothing much I can do about that...

metaron-uk commented:
Rebased to fix a conflict in xbmc/pvr/timers/PVRTimerInfoTag.cpp with latest master.
@ksooo , would it be correct to remove the WIP and RFC flags now?

ryangribble commented:
Yes id think its no longer WIP or RFC.  When do we think this can be merged?  Are we waiting on anything else?

metaron-uk commented:
It's just an @opdenkamp review we're waiting for IMO. I know #7079 was merged without one, resulting in several urgent 'post merge' corrections. I'd rather avoid this if possible. @opdenkamp should be able to merge if he's happy, assuming @ksooo isn't available at that time. 
I'm available to support corrections next week if @opdenkamp's ceiling is still a WIP until then ;).

ksooo commented:
> It's just an @opdenkamp review we're waiting for IMO. I know #7079 was merged without one, resulting in several urgent 'post merge' corrections.

Exactly.

But listen, it is important for me (and my reputation to being able to write larger chunks of 'good' and working code) that the "several urgent 'post merge' corrections" you mentioned were all valid, but just non-functional stuff, no bugs. The bugs were all discovered by other persons, later. Major point of  @opdenkamps's review was the wrong API numbering, which is not unimportant, but has no influence on the functionality. Just being a little picky here...;-)

metaron-uk commented:
@ksooo - Sorry, I didn't intend this as a criticism of you or 7079, just to say that IMO it's important to wait for the code review, even if that means waiting a little longer.

ryangribble commented:
As long as it doesn't mean we "miss" getting into Jarvis, which is why I was asking if there are cut off dates

metaron-uk commented:
I think @ksooo said 20th August for Jarvis alpha 2 cut-off.

ryangribble commented:
Personally i think if no-one has spotted any major issues with this PR so far, rather than waiting for more code reviews, couldnt we merge it and get some actual runtime feedback from real nightly build users?  I feel like we would get more idea on if things are ok or not based on that, plus it gives us more time to rectify anything found as well.

metaron-uk commented:
@opdenkamp , how's your ceiling going?
AFAIK there are no outstanding issues with this PR now. @ksooo has made me get the doxy right and put my spaces in the right places. I think @ryangribble is right about needing runtime feedback next.
Could this be merged now, so I have next week to fix any unexpected problems before alpha 2?

ksooo commented:
@metaron-uk just a quick note as I'm normally not available for Kodi tasks right now. ;-) There is one thing left that definitely needs to be done before merging. So far, nobody has done a code review for the "recording scraper" part of this PR - and from my personal experience this can be a quite sensible topic.

metaron-uk commented:
@ksooo any suggestions about who to ping? I don't know which kodi team members would be appropriate the review this.
My change only affects the pvr:// protocol file name (pvr code), not scraper interface code.
Imho it can't be any worse than before (when year, season and episode weren't included in the pvr:// protocol filename).

metaron-uk commented:
@ronie, would you be able to take a look at the 'scaper support' part of this PR?
It looks like it affects your part of the ship.

1. allow a pvr recording to supply EpisodeName, Year, Season and Episode separate from 'Title'.
2. extend the existing infolabels to get this information from a PVRRecordingInfoTag.
3. add 'EpisodeName' display to confluence skin for recordings.
4. build a pvr:// filename which allows episode information to be scraped using ttvdb.

@xhaggi, I saw you were involved with the introduction if the new info labels for EpisodeName for Isengard. Are you interested in this as well?

ronie commented:
@metaron-uk i'm a skinner, i don't know anything about scrapers or the internal c++ code of kodi.
so #3 i can handle, but for the rest... i have no clue what you're on about :-)

metaron-uk commented:
@ronie - thanks. Hopefully the skin changes are OK.
I'll keep looking for a suitable cpp scraper/infolabel person ...

ronie commented:
> thanks. Hopefully the skin changes are OK.

yup, looks good to me.

metaron-uk commented:
Thanks @ronie.
From 'blame' on  xbmc/GUIInfoManager.cpp and  xbmc/pvr/recordings/PVRRecording.cpp/h which are the other files affected, looks like @xhaggi and @opdenkamp were the most recent team kodi people in that bit of the code.
Will either of you be able to spare time for this in the next few days or so?

xhaggi commented:
just a quick review, also not very familiar with our scraper stuff. I need to trust you that it works because I'm using TVH and it's impossible for me to runtime test it.

metaron-uk commented:
I've incorporated @xhaggi's comments and fixed a bug in the anytime code that must have been introduced during an earlier rebase.
I re-scraped two folders (one with movies, one with star trek) to check the scraper fixes work. They do but the log file is _huge_. I've uploaded a smaller log file (http://xbmclogs.com/pz9ng61ac) which shows me browsing the two folders (you need to scroll to the bottom for the interesting part which shows the file names).

Unfortunately I then tried to re-base on current master. There's something wrong with the libcrossuuid .configure check, to I've reverted to my previous 'master' and am re-compiling to make sure before I push my changes.

mkortstiege commented:
Will check the scraper related changes later today.

ksooo commented:
Let's do a first compilation test... Ofc PVR addons will fail, but Kodi is expected to build fine.

jenkins build this please

ksooo commented:
@mkortstiege could you please take a look at the scraper changes? Thx.

metaron-uk commented:
Build failure was on windows in libmpeg2 (before this change compiled, unfortunately)
If you build addons, I expect pvr.mythtv and pvr.hts to fail without the linked minimal support changes, but the others should pass (only version bump / addon.xml.in changes).

ksooo commented:
This is a final headsup. If nobody objects I will merge the PVR addon changes along with this PR on Aug 20. @opdenkamp Please speak up now or it will happen.

mkortstiege commented:
Since i am not using PVR myself and this PR is not touching the actual scraper code i have no objection.

hudokkow commented:
jenkins build this please

metaron-uk commented:
addon build results as expected - mythtv and hts fail due to missing 'minimal support' changes, but others pass.

ksooo commented:
@metaron-uk merge conflicts! Could you please rebase?

metaron-uk commented:
rebased on current master.

ksooo commented:
@metaron-uk ??? Still seeing merge conflicts.

metaron-uk commented:
#7744 was merged after my last rebase. I tested this combination yesterday so don't see it as an issue. Rebased again.

metaron-uk commented:
Jenkins results lost due to rebase.
Some manual intervention was required during rebase. I also fixed a regression introduced following rebase after f1ddcffe (Implement: Any Time) and the display of series/episode number issue I mentioned in my #7744 comment.
Compiles, runs and does what I expect on my linux x86 box against updated pvr.mythtv and 'minimal support' pvr.hts.
@ksooo, I will be unavailable between 14:30 and ~18:00 (GMT+1) today but should be back after that (Dr permitting) should the flurry of pvr and confluence commits before the alpha 2 deadline require yet another rebase before merge!

ksooo commented:
jenkins build addons please

ksooo commented:
jenkins build this please

ksooo commented:
We're done. Thanks top everybody who made this happen.

metaron-uk commented:
And thanks to you for all the support @ksooo. One step closer to the perfect PVR :+1:
